### PR TITLE
Openimageio, Blender: fix failing builds

### LINKED ISSUE #117038: openimageio fails to build on unstable and latest master
**Describe the bug**

openimageio fails to build on unstable and latest master

**To Reproduce**
```
nix-build -A openimageio
```

[build-log.txt](https://github.com/NixOS/nixpkgs/files/6175844/build-log.txt)

**Notify maintainers**

@goibhniu

### PR DESCRIPTION
**Pull request by illustris**
###### Motivation for this change
On master as of 42ad5a66366314c1f8527620a4568429c933e750 blender and its dependency openimageio fail to build.

###### Things done
- Add a patch to fix failing builds for openimageio 1.8.17 based on https://github.com/OpenImageIO/oiio/pull/2539
- Bump openimageio2 to 2.2.12.0
- Make pugixml a dependency for all targets to fix failing blender build on Linux

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/openimageio/2.x.nix [hunk 1]
```diff
@@ -11,26 +11,20 @@
...
   src = fetchFromGitHub {
     owner = "OpenImageIO";
     repo = "oiio";
     rev = "Release-${version}";
-    sha256 = "1bbxx3bcc5jlb90ffxbk29gb8227097rdr8vg97vj9axw2mjd5si";
+    sha256 = "16z8lnsqhljbfaarfwx9rc95p0a9wxf4p271j6kxdfknjb88p56i";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/openimageio/2.x.nix:28]: Please remove fetchpatch from inputs.

**Comment by illustris** [comment]
Running nixpkgs-review to verify nothing else breaks.

**Comment by illustris** [comment]
appleseed fails to build according to nixpkgs-review, but it is broken on unstable right now anyway.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117128` at eda7d615 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>embree</li>
    <li>embree2</li>
    <li>luxcorerender</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build (new failures):</summary>
  <ul>
    <li><a href="https://gist.github.com/ca23e419be5ae40c8473aa923d1db476">appleseed</a></li></li>
    <li><a href="https://gist.github.com/ef8dab82bc7684865251bb4a32f70dea">natron</a></li></li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/033ed37517203690bc33fb849cc3206f">openimageio</a></li></li>
    <li><a href="https://gist.github.com/83606e2e316d8f24282380e940b9b731">openimageio2</a></li></li>
    <li><a href="https://gist.github.com/eba9901449cd0574c868a45b5c12b808">osl</a></li></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/openimageio/2.x.nix:3:3:
```
  |
3 | , fetchpatch
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/graphics/openimageio/default.nix:1:33:
```
  |
1 | { lib, stdenv, fetchFromGitHub, fetchpatch, boost, cmake, ilmbase, libjpeg, libpng, libtiff
  |                                 ^
```
</li>
  </ul>
</details>

**Review by veprbl — APPROVED** [review]
LGTM

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 117128` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>appleseed: log was empty</li>
  </ul>
</details>
<details>
<summary>
8 packages built:
</summary>
<ul>
<li>blender</li>
<li>embree</li>
<li>embree2</li>
<li>luxcorerender</li>
<li>natron</li>
<li>openimageio</li>
<li>openimageio2</li>
<li>osl</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
blender:
<details>


warning: build-tools-in-build-inputs
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/misc/blender/default.nix:37:3:

```
   |
37 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
</details>

**Comment by illustris** [comment]
@SuperSandro2000 I've moved makeWrapper to nativeBuildInputs

**Pull request merged**